### ISSUE TITLE
Pyright/Pylance inconsistency: Unknown import symbol sklearn

### ISSUE THREAD

**Issue by einarwar** [microsoft/pylance-release#6410]
<h3> Environment data </h3>

-   Language Server version: 2024.9.1
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.12.6
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic
-   scikit-learn version: 1.5.2
-   pyright version: 1.1.380
- Runs in a devcontainer

I have been having an annoyance for some time, where pylance and pyright behaves inconsistently. 

```python
from sklearn.metrics import root_mean_squared_error
```

Pyright does not flag this as an issue, while pylance gives the error: 
```
"root_mean_squared_error" is unknown import symbolPylance[reportAttributeAccessIssue](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportAttributeAccessIssue)
```

When i select `sklearn.metrics -> go to definition` using the python environment where sklearn is installed, i get to `site-packages/sklearn/metrics/__init__.py` where `root_mean_squared_error` correctly is.

If i however select a python interpreter without sklearn installed, when i go to the definition, i am directed to `home/vscode/.vscode-server/extensions/ms-python.vscode-pylance-2024.9.1/dist/bundled/stubs/sklearn/metrics/__init__.pyi`, where the function `root_mean_squared_error` does not exist. 

<h3> Expected behavior </h3>

Would expect pylance to behave as pyright in this case, i.e give no error

<h3> Actual behavior </h3>

Pylance raises an error, while pyright doesn't

<h3> Logs </h3>

When checking the output from Python Language Server, there are two entries that might be related:
```
2024-09-18 19:22:34.475 [info] [Info  - 7:22:34 PM] (31165) [BG(1)] Long operation: checking: file:///usr/local/lib/python3.12/site-packages/sklearn/metrics/__init__.py (4947ms)
2024-09-18 19:22:34.481 [info] [Info  - 7:22:34 PM] (31165) [BG(1)] Long operation: analyzing: file:///usr/local/lib/python3.12/site-packages/sklearn/metrics/__init__.py (4956ms)

```

**Comment by rchiodo** [comment]
Thanks for the issue. This is because our bundled type stubs don't match your scikit-learn version. The stubs here need to be updated:
https://github.com/microsoft/python-type-stubs/tree/main/stubs/sklearn

**Comment by einarwar** [comment]
Thanks for the quick response. Should i raise an issue in that repo as well?

**Comment by rchiodo** [comment]
Yeah it wouldn't hurt. Somebody might have the time to update those stubs.

**Comment by einarwar** [comment]
Took the matter into my own hands: https://github.com/microsoft/python-type-stubs/pull/317

One question though: Is this expected behaviour by pylance? I have read about the differences between pylance and pyright, e.g `useLibraryCodeForTypes`. I tried with this setting both enabled and disabled, without any effect. I would have thought that the default behaviour regardless would be to inspect the package if it was installed. However, `scikit-learn` does not have a `py.typed` file, which might have an effect here

**Comment by debonte** [comment]
> Took the matter into my own hands: [microsoft/python-type-stubs#317](https://github.com/microsoft/python-type-stubs/pull/317)

Thanks @einarwar! I just merged your PR. It will show up in next week's prerelease build.



> `useLibraryCodeForTypes`. I tried with this setting both enabled and disabled, without any effect.

`useLibraryCodeForType` only comes into play when no stubs are found. You can read more about our resolution order [here](https://microsoft.github.io/pyright/#/import-resolution).

**Comment by StellaHuang95** [comment]
This issue has been fixed in prerelease version 2024.9.102, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20249102-26-September-2024-PreRelease)

**Comment by AhmetEkiz** [comment]
With VSCode and Pylance I got errors like the one above that shouldn't have occurred. I just downgraded Pylance and now it works fine.

In my case, working versions of Pylance:

`2024.7.1`
`2024.8.1`
